Source/Core/Editor/UI/Drawers/ProcessSceneReferenceDrawer.cs

```diff
@@ -173,9 +173,9 @@ private void HandleDroppedGameObject(Action<object> changeValueCallback, Process
                     };
 
                     // availableGroups the processSceneObject.Guid and all groups of the PSO
-                    IEnumerable<SceneObjectGroups.SceneObjectGroup> availableGroups = new List<SceneObjectGroups.SceneObjectGroup>() { new SceneObjectGroups.SceneObjectGroup(SceneObjectGroups.UniqueGuidName, processSceneObject.Guid) };
+                    IEnumerable<SceneObjectGroups.SceneObjectGroup> availableGroups = new List<SceneObjectGroups.SceneObjectGroup>() { new SceneObjectGroups.SceneObjectGroup(processSceneObject.gameObject.name, processSceneObject.Guid) };
                     availableGroups = availableGroups.Concat(SceneObjectGroups.Instance.Groups.Where(group => processSceneObject.Guids.Contains(group.Guid) == true));
-                    DrawSearchableGroupListPopup(dropDownRect, onItemSelected, availableGroups);
+                    DrawSearchableGroupListPopup(dropDownRect, onItemSelected, availableGroups, firstItemIsProcessSceneObject: true);
```

Comment:
If I understood correctly, `firstItemIsProcessSceneObject` additionally lists the PSO in the group before listing the available groups. If that's the case, I'd rename it to something more fitting like for example `includeProcessSceneObject`.

Reply:
I had something like `includeProcessSceneObject` at first and then renamed it because I felt I have to be clear that for the implementation inside `SearchableGroupListPopup.SetAvailableGroups(...)`, it has to be the first item in the list.

The correct thing would be of course not to put a PSO in a GroupList. But to have a list of Guids containing groups and a PSO. But I did not wanted to go down that rabbit whole :-/

Reply:
No, but why is it important to mention it's the first item in the list?

Reply:
There is no easy way to distinguish between a fake and real group. If its not the first item then SetAvailableGroups would break.

```
        /// <summary>
        /// Set the groups to be displayed in the list.
        /// </summary>
        /// <param name="availableGroups"></param>
        public void SetAvailableGroups(IEnumerable<SceneObjectGroups.SceneObjectGroup> availableGroups, bool firstItemIsProcessSceneObject = false)
        {
            if (firstItemIsProcessSceneObject)
            {
                this.firstIsProcessSceneObject = firstItemIsProcessSceneObject;
                // remove first item of availableGroups, then sort by Label
                var sortedGroups = availableGroups.Skip(1).OrderBy(t => t.Label).ToList();
                // add the first item back to the beginning of the list
                sortedGroups.Insert(0, availableGroups.First());
                groups = sortedGroups;

            }
            else
            {
                groups = availableGroups.OrderBy(t => t.Label).ToList();
            }

        }
```